tox.ini

```diff
@@ -54,3 +28,14 @@ commands =
     pytest -v {posargs}
 allowlist_externals =
     make
+
+[testenv:black]
+skip_install = true
+deps = pre-commit
+commands = pre-commit run black --all-files
+
+[testenv:linters]
+basepython = python3.10
+skip_install = true
+deps = pre-commit
+commands = pre-commit run --all-files --show-diff-on-failure
```

Comment:
Always add newline at end of file

Reply:
Haha, too bad it doesn't do that automatically 😉